packages/ozone/src/daemon/blob-diverter.ts

```diff
@@ -0,0 +1,229 @@
+  }
+
+  async destroy() {
+    this.destroyed = true
+    const destroyState = (state: PollState) => {
+      if (state.timer) {
+        clearTimeout(state.timer)
+      }
+      return state.promise
+    }
+    await destroyState(this.pollState)
+  }
+
+  async divertBlob() {
+    const toPush = await this.db.db
+      .selectFrom('blob_divert_event')
+      .select('id')
+      .forUpdate()
+      .skipLocked()
+      .where('divertedAt', 'is', null)
+      .where('attempts', '<', 10)
+      .execute()
+    await Promise.all(toPush.map((evt) => this.attemptBlobDivert(evt.id)))
+  }
+
+  private async getBlob({
+    pds,
+    did,
+    cid,
+  }: {
+    pds: string
+    did: string
+    cid: string
+  }) {
+    const blobResponse = await axios.get(
+      `${pds}/xrpc/com.atproto.sync.getBlob`,
+      {
+        params: { did, cid },
+        decompress: true,
+        responseType: 'stream',
+        timeout: 5000, // 5sec of inactivity on the connection
+      },
+    )
+    const imageStream: Readable = blobResponse.data
+    const verifyCid = new VerifyCidTransform(CID.parse(cid))
+    forwardStreamErrors(imageStream, verifyCid)
+
+    return {
+      contentType:
+        blobResponse.headers['content-type'] || 'application/octet-stream',
+      imageStream: imageStream.pipe(verifyCid),
+    }
+  }
+
+  private async uploadBlob(
+    {
+      imageStream,
+      contentType,
+    }: { imageStream: Readable; contentType: string },
+    { subjectDid, subjectUri }: { subjectDid: string; subjectUri: string },
+  ) {
+    if (!this.serviceConfig.authToken || !this.serviceConfig.url) {
+      return false
+    }
+
+    const url = `${this.serviceConfig.url}?did=${subjectDid}&uri=${subjectUri}`
+    const result = await axios(url, {
+      method: 'POST',
+      data: imageStream,
+      headers: {
+        Authorization: this.serviceConfig.authToken,
+        'Content-Type': contentType,
+      },
+    })
+
+    return result.status === 200
+  }
+
+  private async uploadBlobOnService({
+    subjectDid,
+    subjectUri,
+    subjectBlobCid,
+  }: {
+    subjectDid: string
+    subjectUri: string
+    subjectBlobCid: string
+  }): Promise<boolean> {
+    try {
+      if (!this.serviceConfig.authToken || !this.serviceConfig.url) {
+        throw new Error('Blob divert service not configured')
+      }
+
+      const didDoc = await this.idResolver.did.resolve(subjectDid)
+
+      if (!didDoc) {
+        throw new Error('Error resolving DID')
+      }
+
+      const pds = getPdsEndpoint(didDoc)
+
+      if (!pds) {
+        throw new Error('Error resolving PDS')
+      }
+
+      const { imageStream, contentType } = await retryHttp(() =>
+        this.getBlob({ pds, did: subjectDid, cid: subjectBlobCid }),
+      )
+
+      const uploadResult = await retryHttp(() =>
+        this.uploadBlob(
+          { imageStream, contentType },
+          { subjectDid, subjectUri },
+        ),
+      )
```

Comment:
The `imageStream` is stateful so it's not going to be super retryable if it gets partially used-up.  It's possible that  `this.getBlob()` and `this.uploadBlob()` should go in the same retry block.